lib/membrane/rtp/parser.ex

```diff
@@ -42,45 +42,53 @@ defmodule Membrane.RTP.Parser do
               ]
 
   def_input_pad :input,
-    caps: {RemoteStream, type: :packetized, content_format: one_of([nil, RTP, RTCP])},
+    accepted_format:
+      %RemoteStream{type: :packetized, content_format: cf} when cf in [nil, RTP, RTCP],
     demand_mode: :auto
 
-  def_output_pad :output, caps: RTP, demand_mode: :auto
+  def_output_pad :output, accepted_format: RTP, demand_mode: :auto
 
   def_output_pad :rtcp_output,
     mode: :push,
-    caps: {RemoteStream, content_format: RTCP, type: :packetized},
+    accepted_format: %RemoteStream{content_format: RTCP, type: :packetized},
     availability: :on_request
 
   @impl true
-  def handle_init(opts) do
-    {:ok, %{rtcp_output_pad: nil, secure?: opts.secure?}}
+  def handle_init(_ctx, opts) do
+    {[], %{rtcp_output_pad: nil, secure?: opts.secure?}}
   end
 
   @impl true
-  def handle_caps(:input, _caps, _ctx, state) do
-    {{:ok, caps: {:output, %RTP{}}}, state}
+  def handle_stream_format(:input, _stream_format, _ctx, state) do
+    {[stream_format: {:output, %RTP{}}], state}
   end
 
   @impl true
   def handle_pad_added(Pad.ref(:rtcp_output, _ref) = pad, %{playback_state: :playing}, state) do
-    caps = {:caps, {pad, %RemoteStream{content_format: RTCP, type: :packetized}}}
-    {{:ok, [caps]}, %{state | rtcp_output_pad: pad}}
+    actions = [stream_format: {pad, %RemoteStream{content_format: RTCP, type: :packetized}}]
+    {actions, %{state | rtcp_output_pad: pad}}
```

Comment:
I am not sure, `:rtcp_output` is dynamic pad, so you can have multiple `:rtcp_output` pads in a single moment, but `:rtcp_output_pad` field in `state` holds reference to the recently added pad

Reply:
This is rather a workaround for Membrane API to allow leaving this pad unlinked, but at most one instance is ever linked. Notice that if there's more than one `:rtcp_output_pad`, only one would work and that seems to be a bug.
1. Core could consider adding a pad that is like static, but allowed to be ignored
2. This element could raise on linking second `:rtcp_output` for now
3. Otherwise it should properly handle more than one `:rtcp_output`

Getting back to the comment - now I get that using `ctx.pads` might be not that convenient without storing the ref as instead of `ctx.pads[{:rtcp_output, ref}]` you'd have to do `Enum.find` over `ctx.pads`.

Reply:
Not sure what's the full context, but raising when another instance is linked is the way in this case. We could add a feature to the core to handle it but I'm not sure it's worth the effort.

Reply:
@bblaszkow06 `Enum.find` is unnecessary, we can do match like `%{pads: %{Pad.ref(:rtcp_output, ref) => pad}} = ctx`

Reply:
But still, holding this pad in `state` is nicer then performing such a match